mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/presenter/HistoryPresenter.java

```diff
@@ -90,6 +95,37 @@ public void handleTransactionClick(int transactionIndex) {
         mHistoryView.showTransactionDetailDialog(transactionIndex, accountNumber);
     }
 
+    public List<Transaction> getFilteredTransactions(List<Transaction> transactions,
+                                                     List<CheckBoxStatus> statusList) {
+        ArrayList<Transaction> filteredTransactions = new ArrayList<>();
+        List<CheckBoxStatus> CheckBoxStatusList = getCheckedStatusList(statusList);
+        for (Transaction transaction : transactions) {
+            for (CheckBoxStatus status : CheckBoxStatusList) {
+                if (status.getStatus().compareTo(Constants.CREDIT) == 0 &&
```

Comment:
can use `.equals` instead of compareTo (we don't have a custom comparator anyway in CheckBoxStatus)